### PR TITLE
[xwindow] part 6

### PR DESCRIPTION
**Pull request by Neumann-A**

### CODE REVIEW

**Comment by talregev** [comment]
@Neumann-A Can you add `libxtst`?

File: ports/libdrm/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+vcpkg_install_meson()
+vcpkg_fixup_pkgconfig()
+
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
+
+# Handle copyright
+file(MAKE_DIRECTORY "${CURRENT_PACKAGES_DIR}/share/${PORT}/")
+file(TOUCH "${CURRENT_PACKAGES_DIR}/share/${PORT}/copyright") #No license in source?
+endif()
```
Comment (Adela0814) [inline ports/libdrm/portfile.cmake:28]: Please add a new line at the end.

**Review by Adela0814 — CHANGES_REQUESTED** [review]

**Comment by Adela0814** [comment]
Please fix the conflicts.

**Comment by Neumann-A** [comment]
@Adela0814 remove requires:author-response.

**Review by Adela0814 — DISMISSED** [review]

File: ports/libdrm/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+)
+vcpkg_install_meson()
+vcpkg_fixup_pkgconfig()
+
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
+
+# Handle copyright
+file(MAKE_DIRECTORY "${CURRENT_PACKAGES_DIR}/share/${PORT}/")
+file(TOUCH "${CURRENT_PACKAGES_DIR}/share/${PORT}/copyright") #No license in source?
```
Comment (vicroms) [inline ports/libdrm/portfile.cmake:27]: We could try to extract the copyright from the header files, I'll ask for opinions on the team

**Review by vicroms — DISMISSED** [review]

File: ports/libdrm/portfile.cmake (hunk 2 above)
Comment (Neumann-A) [reply ports/libdrm/portfile.cmake:27]: > We could try to extract the copyright from the header files, I'll ask for opinions on the team

Seems like every file has a slightly different copyright owner but they seem to be all MIT.

**Comment by vicroms** [comment]
I think we should find a better mechanism to mark and replace system dependencies.
 
In my opinion all packages with `X_VCPKG_FORCE_X_LIBRARIES` should be empty libraries, and instead of controlling behavior through a triplet variable it should be done through an overlay.

Maybe we could add a new feature to declare system/external dependencies without having to create empty ports? 

Something like:
```
"name": "foo",
"dependencies": [
    {
      "name": "libdrm",
      "external": true,
      "features": [ ... ]
    }
]
```
Then vcpkg can load overlay for these dependencies or continue without them if none is provided.

**Comment by Neumann-A** [comment]
> In my opinion all packages with X_VCPKG_FORCE_X_LIBRARIES should be empty libraries, and instead of controlling behavior through a triplet variable it should be done through an overlay.
> Maybe we could add a new feature to declare system/external dependencies without having to create empty ports?

There should be a triplet variable called `VCPKG_SYSTEM_PROVIDED_PORTS` which is a list of ports which vcpkg treats as system provided and simply skips the install step (just like empty ports). No need to force people to c&p the same empty overlay all day long. 
Of course there also needs to be the invers. Meaning some json field in the manifest to declare a port (not a dependency) as normally provided by the system with an option to opt out of the behavior. (`VCPKG_PROVIDED_SYSTEM_PORTS` ?)

**Comment by Neumann-A** [comment]
The above being said: The current approach has one benefit since you could add additional messaging or testing if the libraries provided by the system actually is provided by the system. So basically the system part of the port could have code (a `CMakeLists.txt`) which actually tries to check if it can find the library or package required.

**Comment by BillyONeal** [comment]
> In my opinion all packages with `X_VCPKG_FORCE_X_LIBRARIES` should be empty libraries, and instead of controlling behavior through a triplet variable it should be done through an overlay.

I agree something in this direction is the "correct" thing, but given that all these X things have already been merged with this form, I'm inclined to follow the pattern already followed elsewhere, unless this is an over-my-dead-body issue for you @vicroms?

**Comment by BillyONeal** [comment]
I discussed this over chat with @vicroms and he pointed out this is different than the other "X" PRs because it leaves these ports completely untested. They become empty on non-Windows, and they have `  "supports": "!windows"` or `"supports": "!(windows | osx)"`

**Comment by Neumann-A** [comment]
> "X" PRs because it leaves these ports completely untested. 

Yeah, I didn't want to force a world rebuild for this one by adding the variable to the linux triplet. I could also comment out the if for testing it once?

File: ports/libdrm/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,28 @@
+if(NOT X_VCPKG_FORCE_VCPKG_X_LIBRARIES AND NOT VCPKG_TARGET_IS_WINDOWS)
```
Comment (Neumann-A) [inline ports/libdrm/portfile.cmake:1]:
```suggestion
if(0)

```

File: ports/libxshmfence/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,32 @@
+if(NOT X_VCPKG_FORCE_VCPKG_X_LIBRARIES AND NOT VCPKG_TARGET_IS_WINDOWS)
```
Comment (Neumann-A) [inline ports/libxshmfence/portfile.cmake:1]:
```suggestion
if(0)

```

**Comment by Neumann-A** [comment]
@BillyONeal The ports were tested in [4801462f8c9c723c178a2e4ee9de65c883d69f5c](https://dev.azure.com/vcpkg/public/_build/results?buildId=79659&view=logs&jobId=afe306fb-88a0-5d5e-2934-71aa01507316): linux/osx green

**Comment by Adela0814** [comment]
Please fix the conflicts.

**Comment by BillyONeal** [comment]
> I discussed this over chat with @vicroms and he pointed out this is different than the other "X" PRs because it leaves these ports completely untested. They become empty on non-Windows, and they have ` "supports": "!windows"` or `"supports": "!(windows | osx)"`

Sorry, put the tag back because

**Comment by JavierMatosD** [comment]
Closing this PR on account that nothing here is tested. Feel free to reopen if you still want this and can provide justification.

**Pull request closed without merging**